gobblin-runtime/src/main/java/gobblin/runtime/TaskStateCollectorService.java

```diff
@@ -160,6 +190,17 @@ public Void call() throws Exception {
       this.jobState.addTaskState(taskState);
     }
 
+    // Finish any addtional steps defined in handler globally.
+    // Currently implemented handler for Hive registration only.
+    if (optionalTaskCollectorHandler.isPresent() &&
+        this.jobState.getPropAsBoolean(ConfigurationKeys.PIPELINED_HIVE_REGISTRATION_ENABLED)){
+      LOGGER.info("Enabled Pipelined HiveRegistrationPublisher for " + taskStateQueue.size() + " tasks");
+
+      long startTime = System.currentTimeMillis();
+      ((HiveRegistrationPublisher)(this.optionalTaskCollectorHandler.get())).publishData(taskStateQueue);
```

Comment:
Please create a handler class or interface since the cast here means the code can't invoke the handler generically.